src/v/archival/archiver_operations_api.h

```diff
@@ -0,0 +1,252 @@
+/*
+ * Copyright 2024 Redpanda Data, Inc.
+ *
+ * Licensed as a Redpanda Enterprise file under the Redpanda Community
+ * License (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * https://github.com/redpanda-data/redpanda/blob/master/licenses/rcl.md
+ */
+
+#pragma once
+
+#include "archival/async_data_uploader.h"
```

Comment:
It seems like most of the functions could be decoupled from cluster/archival, like find_upload_candidates, schedule_uploads, and upload_manifest. I suppose it's still possible for the guts of those methods to live elsewhere, even with this interface.

Reply:
Yes, but why would we want this? It shouldn't go into `cloud_storage` in my opinion because it's only used in the write path (for which we have `archival` namespace).

Reply:
If these are decoupled, they become easier to test independently of one another, without having to depend things that are required for the rest of the interface. IMO it's also much easier to reason about and read when the interfaces are tightly scoped, though this PR already does a good job of detailing the interface.

Reply:
> The original comment though was just curiosity of what is being used from async_data_uploader.h?

It's used in the next commits (I blame git absorb).